oshmem/mca/sshmem/verbs/configure.m4

```diff
@@ -37,6 +37,49 @@ AC_DEFUN([MCA_oshmem_sshmem_verbs_CONFIG],[
     AC_SUBST([oshmem_verbs_LDFLAGS])
     AC_SUBST([oshmem_verbs_LIBS])
 
+    oshmem_have_exp_reg_shared_mr=0
+    AS_IF(
+        [test "$oshmem_verbs_sm_build_verbs" = "1"],
+        [
+            OSHMEM_LIBSHMEM_EXTRA_LDFLAGS="$OSHMEM_LIBSHMEM_EXTRA_LDFLAGS $oshmem_verbs_LDFLAGS"
+            OSHMEM_LIBSHMEM_EXTRA_LIBS="$OSHMEM_LIBSHMEM_EXTRA_LIBS $oshmem_verbs_LIBS"
+
+            oshmem_verbs_save_CFLAGS="$CFLAGS"
+            oshmem_verbs_save_CPPFLAGS="$CPPFLAGS"
+            oshmem_verbs_save_LDFLAGS="$LDFLAGS"
+            oshmem_verbs_save_LIBS="$LIBS"
+
+            CFLAGS="-Werror"
+            CPPFLAGS="$CPPFLAGS $oshmem_verbs_CPPFLAGS"
+            LDFLAGS="$LDFLAGS $oshmem_verbs_LDFLAGS"
+            LIBS="$LIBS $oshmem_verbs_LIBS"
+
+            AC_COMPILE_IFELSE(
+                        [AC_LANG_PROGRAM([[#include <infiniband/verbs_exp.h>]],
+                                         [[
+                                             struct ibv_exp_reg_shared_mr_in in_smr;
+                                             uint64_t access_flags = IBV_EXP_ACCESS_SHARED_MR_USER_READ   |
+                                                                     IBV_EXP_ACCESS_SHARED_MR_USER_WRITE  |
+                                                                     IBV_EXP_ACCESS_SHARED_MR_GROUP_READ  |
+                                                                     IBV_EXP_ACCESS_SHARED_MR_GROUP_WRITE |
+                                                                     IBV_EXP_ACCESS_SHARED_MR_OTHER_READ  |
+                                                                     IBV_EXP_ACCESS_SHARED_MR_OTHER_WRITE;
+                                             in_smr.exp_access = access_flags;
+                                             ibv_exp_reg_shared_mr(&in_smr);
+                                         ]])],
+                        [oshmem_have_exp_reg_shared_mr=1],
+                        [oshmem_have_exp_reg_shared_mr=0])
```

Comment:
can we set `oshmem_verbs_sm_build_verbs=0` directly from here?

Reply:
yes, but this is how it is also done in the next check in this file so i kept the consistency